Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,355 @@
+                return InValidOutput;
+            }
+
+            var HighestBalance = decimal.MinValue;
+            var RichestPerson = new StringBuilder();
+
+            for (int i = 0; i <= peopleAndBalances.Length - 1; i++)
+            {
+                var currentPersonData = peopleAndBalances[i].Split(',');
+                var isValid = decimal.TryParse(currentPersonData[currentPersonData.Length - 1], out decimal highestAmountOfPerson);
+
+                if (!isValid)
+                {
+                    return "N/A.";
+                }
+
+                HighestBalance = FindRichestPerson(peopleAndBalances, HighestBalance, RichestPerson, i, currentPersonData, highestAmountOfPerson);
+
+            }
+
+            var answer = $"{RichestPerson.ToString()} is the richest person. ¤{HighestBalance}.";
+
+            return answer;
+
+
+        }
+
+        private static decimal FindRichestPerson(string[] peopleAndBalances, decimal HighestBalance, StringBuilder RichestPerson, int i, string[] currentPersonData, decimal highestAmountOfPerson)
+        {
+            if (highestAmountOfPerson > HighestBalance)
+            {
+                HighestBalance = highestAmountOfPerson;
+                var currentOne = RichestPerson.ToString();
+
+                if (string.IsNullOrEmpty(currentOne))
+                {
+                    RichestPerson.Append(currentPersonData[0]);
+                }
+                else
+                {
+                    RichestPerson.Replace(currentOne, currentPersonData[0]);
+                }
+
+
+            }
+            else if (highestAmountOfPerson == HighestBalance)
+            {
+                if (i == peopleAndBalances.Length - 1)
+                {
+                    RichestPerson.Append(" and ");
+                }
+                else
+                {
+                    RichestPerson.Append(", ");
+                }
+
+                RichestPerson.Append(currentPersonData[0]);
+            }
+
+            return HighestBalance;
+        }
+
+        private static void FindMorePeopleWhicAreEvenPoor(string[] peopleAndBalances, decimal HighestBalance, StringBuilder RichestPerson, int i, string[] currentPersonData, decimal highestAmountOfPerson)
+        {
+            if (highestAmountOfPerson == HighestBalance)
+            {
+                if (i == peopleAndBalances.Length - 1)
+                {
+                    RichestPerson.Append(" and ");
+                }
+                else
+                {
+                    RichestPerson.Append(", ");
+                }
+
+                RichestPerson.Append(currentPersonData[0]);
+            }
+        }
+
+        private static decimal FindRichestPerson(decimal HighestBalance, StringBuilder RichestPerson, string[] currentPersonData, decimal highestAmountOfPerson)
+        {
+            if (highestAmountOfPerson > HighestBalance)
+            {
+                HighestBalance = highestAmountOfPerson;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Introduce a new variable instead of reusing the parameter 'HighestBalance'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4937470)

Reply:
Looks legal to me so I do not understand why this is wrong